pkg/executor/utils.go

```diff
@@ -119,3 +121,70 @@ func encodePassword(u *ast.UserSpec, authPlugin *extension.AuthPlugin) (string,
 	}
 	return u.EncodedPassword()
 }
+
+var taskPool = sync.Pool{
+	New: func() any { return &gotask{} },
+}
+
+type gotask struct {
+	f    func()
+	next *gotask
+}
+
+type gopool struct {
```

Comment:
There is some `GoroutinePool` like structures in other packages, how about renaming it to `IndexLookupWorkerPool` or `CopWorkerPool` it we do not want it to be a generic used type?

Reply:
Actually I prefer to keep it simple and private currently. [gp](https://github.com/tiancaiamao/gp) and [pool](https://github.com/pingcap/tidb/tree/master/pkg/resourcemanager/pool) have already been used in tidb, but neither of them meet my requirement. So I just implement a tiny one here and maybe we can extract uniform GoroutinePool interface in the future.